tests/wallet/conftest.py

```diff
@@ -2,18 +2,85 @@
 
 from contextlib import AsyncExitStack
 from dataclasses import replace
-from typing import Any, AsyncIterator, Dict, List
+from typing import Any, AsyncIterator, Awaitable, Callable, Dict, List, Literal, Tuple
 
 import pytest
 
 from chia.consensus.constants import ConsensusConstants
+from chia.consensus.cost_calculator import NPCResult
+from chia.full_node.full_node import FullNode
 from chia.rpc.wallet_rpc_client import WalletRpcClient
+from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.types.peer_info import PeerInfo
 from chia.util.ints import uint32, uint64, uint128
 from chia.wallet.util.tx_config import DEFAULT_TX_CONFIG, TXConfig
 from chia.wallet.wallet_node import Balance
+from chia.wallet.wallet_state_manager import WalletStateManager
 from tests.environments.wallet import WalletEnvironment, WalletState, WalletTestFramework
 from tests.util.setup_nodes import setup_simulators_and_wallets_service
+from tests.wallet.wallet_block_tools import WalletBlockTools
+
+
+@pytest.fixture(scope="function", autouse=True)
+def block_is_current_at(monkeypatch: pytest.MonkeyPatch) -> None:
+    def make_new_synced(func: Callable[..., Awaitable[bool]]) -> Any:
+        async def mocked_synced(self: Any, block_is_current_at: int = 0) -> bool:
+            return await func(self, block_is_current_at)
+
+        return mocked_synced
+
+    monkeypatch.setattr(WalletStateManager, "synced", make_new_synced(WalletStateManager.synced))
+    monkeypatch.setattr(FullNode, "synced", make_new_synced(FullNode.synced))
+
+
+@pytest.fixture(scope="function", autouse=True)
+async def ignore_block_validation(request: pytest.FixtureRequest, monkeypatch: pytest.MonkeyPatch) -> None:
```

Comment:
I can see, two years from now, somebody coming across this code and think: "I can simplify all this code by making the wallet tests use the same `BlockTools` as all the other tests.

I think there should be a comment here, explaining why that's a bad idea.